src/slack-commands.ts

```diff
@@ -0,0 +1,151 @@
+import {
+    BlockElementAction,
+    DialogSubmitAction,
+    InteractiveAction,
+    RespondArguments,
+    WorkflowStepEdit,
+} from "@slack/bolt";
+import {emojiDescriptions, EmojiKey, EmojiStore} from "./emoji";
+import {PlainTextOption} from "@slack/types";
+
+type Action = DialogSubmitAction | WorkflowStepEdit | InteractiveAction | BlockElementAction
+
+export async function handleAction(action: Action): Promise<ActionResult | undefined> {
+    if (action.type === 'static_select') {
+        if (action.action_id === 'configure_emoji') {
+            const newEmoji = action.block_id
+            const forWhich = action.selected_option.value as EmojiKey
+            await EmojiStore.real().set(forWhich, newEmoji)
+            return {
+                type: "configure_emoji",
+                newValue: newEmoji,
+            }
+        }
+    } else {
+        console.log("unknown action: ", action.type)
+    }
+    return
+}
+
+type ActionResult =
+    | ConfigureEmojiResult
+
+interface ConfigureEmojiResult {
+    type: 'configure_emoji'
+    newValue: string,
+}
+
+/**
+ * Parses a command that starts with "emoji".
+ *
+ * The command must either be just "emoji", or else contain 1 or more images of the form ":image:". It may not contain
+ * anything else.
+ * @param command
+ */
+export function parseEmojiCommand(command: String): {image: string} | {err: string} | 'describe' {
+    const fullLine = command.match(/^emoji(?:\s+(?<images>:[^:]+:)?\s*(?<extra>.*))?/)
+    if (fullLine === null) {
+        return {err: "Unrecognized command"} // not expected!
+    }
+    if ((fullLine.groups?.extra ?? "").length > 0) {
```

Comment:
nit: don't need the `?? ""` because `undefined > 0 == false`

Reply:
Removing it gave me a warning that `fullLine.groups?.extra?.length` is undefined — I guess typescript (or maybe just intellij's default linting?) doesn't like the implicit `undefined > 0 is falsey` handling. `TS2532`. 🤷‍♂️ Sounds like it's not a big deal from your perspective, so I'll keep it as is.